### PR TITLE
pythonPackages.ntlm-auth: 1.0.3 -> 1.4.0 to fix build

### PR DESCRIPTION
**Pull request by bhipple**
On master and 20.03, this is failing to build on `python 3.8`.

https://hydra.nixos.org/build/115517329
https://hydra.nixos.org/build/114714922

CC @NixOS/nixos-release-managers
ZHF: #80379

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bhipple** [comment]
```
[9 built (1 failed), 5 copied (0.1 MiB), 0.1 MiB DL]
error: build of '/nix/store/fkxj9jzlxkm5jrhm3sqfmjc46xgzpy0w-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/83560
1 package marked as broken and skipped:
python27Packages.resampy

1 package failed to build:
fluidasserts

12 package built:
python27Packages.IBMQuantumExperience python27Packages.ntlm-auth python27Packages.pywinrm python27Packages.requests_ntlm python37Packages.IBMQuantumExperience python37Packages.ntlm-auth python37Packages.pywinrm python37Packages.qiskit-ibmq-provider python37Packages.requests_ntlm python38Packages.IBMQuantumExperience python38Packages.ntlm-auth python38Packages.requests_ntlm
```
`fluidasserts` appears to be broken on master due to unrelated `protobuf` issues.

**Comment by worldofpeace** [comment]
What was changed in this update, are there changelogs? `1.0.3 -> 1.4.0` could be a big update. It might be ok if it is a leaf package, but maybe we could use a patch to fix it for 3.8.

**Comment by bhipple** [comment]
The changelog looks relatively innocuous and does not appear to have any breaking changes:
https://github.com/jborean93/ntlm-auth/releases

I've built all the reverse dependencies successfully with `nixpkgs-review` above, and they pass their build/test -- but I don't actually use this pkg personally and haven't done any manual end-to-end integration testing.

File: pkgs/development/python-modules/ntlm-auth/default.nix [hunk 1]
```diff
@@ -28,7 +29,7 @@ buildPythonPackage rec {
 
   meta = with lib; {
     description = "Calculates NTLM Authentication codes";
-    homepage = https://github.com/jborean93/ntlm-auth;
+    homepage = "https://github.com/jborean93/ntlm-auth";
     license = licenses.lgpl3;
```
Comment (nh2) [inline pkgs/development/python-modules/ntlm-auth/default.nix]:
```suggestion
    license = licenses.mit;
```

From https://github.com/jborean93/ntlm-auth/releases/tag/v1.1.0

> License change to MIT

**Comment by bhipple** [comment]
Good catch on the license; updated.  I also ammended the PR description with links to failing Hydra jobs on trunk and release-20.03, per suggestion from @worldofpeace.

**Comment by nh2** [comment]
In case it is useful, I also ran `nix-review` on the previous 20.03 version of this: https://github.com/NixOS/nixpkgs/pull/83564#issuecomment-605493367

**Review by worldofpeace — APPROVED** [review]
Looks good

**Pull request merged**